### PR TITLE
Right-click-functions

### PR DESCRIPTION
**Pull request by jiteshjha**

### CODE REVIEW

**Comment by kamakshidasan** [comment]
I haven't even gone through it - but you seem so sure. 

Are you sure?

**Comment by jiteshjha** [comment]
Lots of work pending, please don't merge this now.

File: index.html [hunk 1]
```diff
@@ -119,36 +119,39 @@
...
                                 var cellSource = cell.source;
-                                while (cellSource.isEdge() == true) {
-                                    cellSource = cellSource.source;
+                                while(cellSource.isEdge() == true) {
+                                  cellSource = cellSource.source;
                                 }
                                 if (cellSource.value == "ExplicitOutputPort" || cellSource.value == "ExplicitInputPort") {
-                                    style += ';' + 'ExplicitLink';
+                                    if(style == null) {
+                                      style = ';' + 'ExplicitLink' + ';';
```
Comment (kamakshidasan) [inline index.html]: I'm guessing that this semi-colon should be removed?

Comment (jiteshjha) [inline index.html]: The code is uncommented and looks messy now, I'll fix all of this first thing tomorrow.

File: index.html [hunk 2]
```diff
@@ -1212,10 +1252,240 @@
...
+                          option.value = item;
+                          option.text = item;
+                          option.setAttribute("id", item);
+                          newList.appendChild(option);
+                        }
+                      }
+
+                      var selectedFontFamily = 0;
+                      if (style.indexOf('fontFamily') > -1) {
+                          selectedFontFamily =  style.substring(style.lastIndexOf("fontFamily=") + "fontFamily=".length, style.lastIndexOf(";"));
```
Comment (kamakshidasan) [inline index.html]: check this.

File: index.html [hunk 3]
```diff
@@ -1212,10 +1252,240 @@
...
+                graph.getModel().setValue(cell, propertiesObject['text']);
+                wind.destroy();
+            };
+            myform.appendChild(btn);
+
+            // Base heights without fields : 135 px
+            height = 135 + 26 * defaultProperties.length + 15;
+            content.appendChild(myform);
+            var wind = showModalWindow(graph, 'Text and Text font', content, 450, height);
+
```
Comment (kamakshidasan) [inline index.html:1466]: Should you remove #L1455 - #L1487? I'm guessing this is already there in getFontStyle()?

Comment (jiteshjha) [inline index.html:1466]: The code below finds if there's already a fontStyle set for the selected edge label, and toggles the class of the respective button(s). getFontStyle() has nothing to do with this.

**Comment by jiteshjha** [comment]
Made the changes, ready for merge?

File: index.html (hunk 3 above)
Comment (kamakshidasan) [inline index.html:1466]: Then can you a write a function called setFontStyle() and call it here? I feel there is redundancy.

**Comment by kamakshidasan** [comment]
Nope. :P

On Wed, Jun 15, 2016 at 11:02 PM, Jitesh Kumar Jha <notifications@github.com

**Comment by jiteshjha** [comment]
Added the requested changes.

**Comment by kamakshidasan** [comment]
Good work. :)

**Pull request merged**